src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/ExportProviderBuilderTests.cs

```diff
@@ -49,6 +54,98 @@ public async Task MultipleMefCompositionsAreCached()
+
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var assembly = Assert.Single(assemblies, a => a.GetName().Name == assemblyName);
+            var type = Assert.Single(assembly.GetTypes(), t => t.FullName?.Contains("ExportedType") == true);
+            var values = testServer.ExportProvider.GetExportedValues(type, contractName: null);
+            Assert.Single(values);
+        }
+
+        private string GenerateDll(string reservedCharacter, out string assemblyName)
+        {
+            var directory = TempRoot.CreateDirectory();
+            CSharpCompilationOptions options = new(OutputKind.DynamicallyLinkedLibrary);
+
+            // Create a dll that exports and imports a mef type to ensure that MEF attempts to load and create a MEF graph
+            // using this assembly.
+            var source = """
+                namespace MyTestExportNamespace
+                {
+                    [System.ComponentModel.Composition.Export(typeof(ExportedType))]
+                    public class ExportedType { }
+
+                    public class ImportType
+                    {
+                        [System.ComponentModel.Composition.ImportingConstructorAttribute]
+                        public ImportType(ExportedType t) { }
+                    }
+                }
+                """;
+
+            // Generate an assembly name associated with the character we're testing - this ensures
+            // that if multiple of these tests run in the same process we're making sure that the correct expected assembly is loaded.
+            assemblyName = "MyAssembly" + reservedCharacter.GetHashCode();
+#pragma warning disable RS0030 // Do not use banned APIs - intentionally using System.ComponentModel.Composition to verify mef construction.
+            var compilation = CSharpCompilation.Create(
+                assemblyName,
+                syntaxTrees: [SyntaxFactory.ParseSyntaxTree(SourceText.From(source, encoding: null, SourceHashAlgorithms.Default))],
+                references:
+                [
+                    NetStandard20.References.mscorlib,
+                    NetStandard20.References.netstandard,
+                    NetStandard20.References.SystemRuntime,
+                    MetadataReference.CreateFromFile(typeof(System.ComponentModel.Composition.ExportAttribute).Assembly.Location)
+                ],
+                options: options);
+#pragma warning restore RS0030 // Do not use banned APIs
+
+            // Write a dll to a subdir with a reserved character in the name.
+            var tempSubDir = directory.CreateDirectory(reservedCharacter);
+            var tempFile = tempSubDir.CreateFile($"{assemblyName}.dll");
+            var dllData = compilation.EmitToStream();
+            tempFile.WriteAllBytes(dllData.ToArray());
+
+            // Mark the file as read only to prevent mutations.
```

Comment:
Does this break us cleaning up the temp files later?

Reply:
I didn't see issues, but let me check.  I stole this from a compiler test